link/uprobe.go

```diff
@@ -19,7 +20,10 @@ var (
 	haveRefCtrOffsetPMU     = internal.NewFeatureTest("RefCtrOffsetPMU", "4.20", func() error {
 		_, err := os.Stat(uprobeRefCtrOffsetPMUPath)
 		if err != nil {
-			return internal.ErrNotSupported
+			if errors.Is(err, unix.EINVAL) {
```

Comment:
I'm a bit surprised this works to be honest. From `man 2 stat`:

```
ERRORS
       EACCES Search permission is denied for one of the directories in the path prefix of pathname.  (See also path_resolution(7).)

       EBADF  fd is not a valid open file descriptor.

       EBADF  (fstatat()) pathname is relative but dirfd is neither AT_FDCWD nor a valid file descriptor.

       EFAULT Bad address.

       EINVAL (fstatat()) Invalid flag specified in flags.

       ELOOP  Too many symbolic links encountered while traversing the path.

       ENAMETOOLONG
              pathname is too long.

       ENOENT A component of pathname does not exist or is a dangling symbolic link.

       ENOENT pathname is an empty string and AT_EMPTY_PATH was not specified in flags.

       ENOMEM Out of memory (i.e., kernel memory).

       ENOTDIR
              A component of the path prefix of pathname is not a directory.

       ENOTDIR
              (fstatat()) pathname is relative and dirfd is a file descriptor referring to a file other than a directory.

       EOVERFLOW
              pathname or fd refers to a file whose size, inode number, or number of blocks cannot be represented in, respectively, the types off_t, ino_t, or blkcnt_t.  This error can occur when, for  exam‐
              ple, an application compiled on a 32-bit platform without -D_FILE_OFFSET_BITS=64 calls stat() on a file whose size exceeds (1<<31)-1 bytes.
```

From this, I'd expect this to return ENOENT, for which the go equivalent is `os.ErrNotExist`. Shouldn't this be:

```
if errors.Is(err, os.ErrNotExist) { return ErrNotSupported }
```

?

Reply:
> Where did you infer that it works though btw, a bit confused by that statement.

I meant: CI doesn't bail out. The reason is that on CI the path always exists, I think. So we never enter this condition.